mantis-server/mantis-server-agent/src/main/java/io/mantisrx/server/agent/metrics/cgroups/MemorySubsystemProcess.java

```diff
@@ -47,4 +55,19 @@ public void getUsage(UsageBuilder usageBuilder) throws IOException {
             log.warn("stats for memory not found; stats={}", stats);
         }
     }
+
+    private void handleV2(UsageBuilder usageBuilder) throws IOException {
+        Long memLimit = cgroup.getMetric(null, "memory.max");
+        usageBuilder.memLimit(memLimit);
+
+        Map<String, Long> stats = cgroup.getStats(null, "memory.stat");
+        Optional<Long> anon = Optional.ofNullable(stats.get("anon"));
+        Optional<Long> shmem = Optional.ofNullable(stats.get("shmem"));
```

Comment:
In theory with cgroupv2 you could use `memory.current`, but I'm assuming there is good reason to calculate the memory this way (anon+shmem), so I say keep doing that.

Reply:
Thanks for this comment. I've cleaned up the code to use `memory.current` instead of `shmem`.